public/app/features/explore/LogsVolumePanel.tsx

```diff
@@ -46,22 +56,51 @@ function ErrorAlert(props: { error: DataQueryError }) {
   );
 }
 
+type MergedInfo = {
+  logsVolumeData?: DataQueryResponse;
+  logLinesBased: boolean;
+  range: AbsoluteTimeRange;
+};
+
+function mergeFullRangeAndLogsBasedData(
+  logLinesBased: DataQueryResponse | undefined,
+  logLinesBasedVisibleRange: AbsoluteTimeRange | undefined,
+  fullRangeData: DataQueryResponse | undefined,
+  absoluteRange: AbsoluteTimeRange
+): MergedInfo {
+  const logLinesFrames = logLinesBased?.data;
+  if (logLinesFrames && logLinesFrames.length) {
```

Comment:
you're right, i was trying to avoid any functional-changes in this PR, but with https://github.com/grafana/grafana/pull/50676 merged we can be somewhat more brave :) i'll switch it around.

Reply:
ok, switched :)